include/networkit/matching/BSuitorMatcher.hpp

```diff
@@ -0,0 +1,199 @@
+        }
+    }
+
+    void insert(const Node &v) {
+        assert(size < max_size);
+
+        list.emplace_back(v);
+        size += 1;
+        if (self < v.id)
+            weight += v.weight;
+        if (size == max_size && !list.empty()) {
+            min = *std::min_element(list.begin(), list.end(), [](const Node &a, const Node &b) {
+                if (a.weight == b.weight) {
+                    return a.id > b.id;
+                }
+                return a.weight < b.weight;
+            });
+        }
+    }
+
+    void remove(const Node &u) {
+        list.erase(
+            std::remove_if(list.begin(), list.end(), [u](const Node &p) { return p.id == u.id; }),
+            list.end());
+        size -= 1;
+        if (self < u.id)
+            weight -= u.weight;
+        if (u.id == min.id) {
+            min = Node(none, 0);
+        }
+    }
+
+    void sort() {
+        std::sort(list.begin(), list.end(), [](const Node &u, const Node &v) {
+            return (u.weight > v.weight || (u.weight == v.weight && u.id < v.id));
+        });
+    }
+};
+
+struct Suitors : public Data {
+    Suitors(index i, count b) : Data(i, b) {}
+
+    edgeweight getWeight() {
+        edgeweight w = 0.0;
+
+#pragma omp parallel for reduction(+ : w)
+        for (omp_index i = 0; i < static_cast<omp_index>(list.size()); i++) {
+            w += list.at(i).weight;
+        }
+        return w;
+    }
+};
+
+struct Proposed : public Data {
+    Proposed(index i, count b) : Data(i, b) {}
+};
+
+/**
+ * @ingroup matching
+ * B-Suitor matching finding algorithm.
+ */
+class BSuitorMatcher final : public BMatcher {
+public:
+    /**
+     * Computes a 1/2-approximate maximum weight b-matching of an undirected weighted Graph @c G
+     * using the sequential b-Suitor algorithm published by Khan et al. in "Efficient Approximation
+     * Algorithms For Weighted B-Matching", SIAM Journal on Scientific Computing, Vol. 38, Iss. 5
+     * (2016).
+     *
+     * @param G An undirected graph.
+     * @param b A vector of @a b values that represents the max number of edges per vertex @a v in
+     * the b-Matching (b.at(v)).
+     */
+    BSuitorMatcher(const Graph &G, const std::vector<count> &b);
+
+    /**
+     * @param G An undirected graph.
+     * @param b A value @a b that represents the max number of edges per vertex in the b-Matching.
+     * Defaults to the ordinary 1-Matching.
+     */
+    BSuitorMatcher(const Graph &G, count b = 1);
+
+    /**
+     * @param G  An undirected graph.
+     * @param path  A path to a file containing @a b values that represents the max number of edges
+     * per vertex in the b-Matching.
+     */
+    BSuitorMatcher(const Graph &G, const std::string &path);
+
+    ~BSuitorMatcher() override = default;
+
+    /**
+     * Runs the algorithm.
+     */
+    void run() override;
+
+    edgeweight getWeight() const;
+
+    void buildBMatching();
+
+private:
+    std::vector<std::unique_ptr<Suitors>> S;
```

Comment:
I suggest not using single char names for class member variables.

Reply:
[I'll change it](https://github.com/networkit/networkit/pull/1109#discussion_r1356881259).

Reply:
done in cb0aa0e992.